packages/core/src/routes/interaction/utils/sign-in-experience-validation.ts

```diff
@@ -119,3 +122,11 @@ export const profileValidation = (profile: Profile, { signUp }: SignInExperience
     assertThat(signUp.password, forbiddenIdentifierError);
   }
 };
+
+export const getSignInExperience = async (ctx: Context, provider: Provider) => {
+  const interaction = await provider.interactionDetails(ctx.req, ctx.res);
+
+  return getSignInExperienceForApplication(
+    typeof interaction.params.client_id === 'string' ? interaction.params.client_id : undefined
```

Comment:
Why using `typeof foo === 'string'`, any particular reason?

Reply:
that's a legacy code. `interaction.params` type is an unknown object, ` interaction.params.client_id ` type is unknown, use this to guard the client_id string type.